src/devices/Mcp3428/Mcp3428.cs

```diff
@@ -0,0 +1,301 @@
+        }
+
+        protected void ReadConfigByte(byte config)
+        {
+            _isReadyBit = (config & Helpers.Masks.ReadyMask) == 0; // Negated bit
+            _lastChannel = (byte)((config & Helpers.Masks.ChannelMask) >> 5);
+            _mode = (ModeEnum)(config & Helpers.Masks.ModeMask);
+            _pgaGain = (GainEnum)(config & Helpers.Masks.GainMask);
+            _resolution = (ResolutionEnum)(config & Helpers.Masks.ResolutionMask);
+        }
+
+        protected double ReadValue(int channel = -1)
+        {
+            if (Mode == ModeEnum.OneShot)
+            {
+                OneShotRead(channel);
+            }
+            else
+            {
+                if (channel > 0 && channel != LastChannel)
+                {
+                    var conf = SetChannelBits(LastConfigByte, channel);
+                    WriteConfig(conf);
+                }
+
+                using (var source = new CancellationTokenSource(TimeSpan.FromMilliseconds(WaitTime * 5)))
+                {
+                    WaitForConversion(TimeSpan.FromMilliseconds(WaitTime / 5), cancellationToken: source.Token);
+                }
+            }
+
+            //var value = BinaryPrimitives.ReadInt16LittleEndian(_readBuffer.AsSpan().Slice(1, 2));
+            var value = BinaryPrimitives.ReadInt16BigEndian(_readBuffer.AsSpan().Slice(0, 2));
+            LastConversion = new ConversionResult(_lastChannel, value, Resolution);
+            //_lastValue = value * LSBValue(Resolution);
+            return LastConversion.Voltage;
+        }
+
+        /// <summary>
+        /// Write configuration register and read back value
+        /// </summary>
+        /// <param name="channel">The channel.</param>
+        /// <param name="mode">The mode.</param>
+        /// <param name="resolution">The resolution.</param>
+        /// <param name="pgaGain">The pga gain.</param>
+        /// <returns><c>true</c> if all values are set correctly, <c>false</c> otherwise.</returns>
+        /// <autogeneratedoc />
+        protected bool SetConfig(int channel = 0, ModeEnum mode = ModeEnum.Continuous,
+            ResolutionEnum resolution = ResolutionEnum.Bit12, GainEnum pgaGain = GainEnum.X1)
+        {
+            if (channel < 0 || channel > ChannelCount-1) throw new ArgumentOutOfRangeException(nameof(channel));
+            byte conf = 0;
+            var ok = true;
+            conf = SetModeBit(conf, mode);
+            conf = SetChannelBits(conf, channel);
+            conf = SetGainBits(conf, pgaGain);
+            conf = SetResolutionBits(conf, resolution);
+            conf = SetReadyBit(conf, false);
+
+            _i2cDevice.WriteByte(conf);
+
+            _i2cDevice.Read(_readBuffer);
+            ReadConfigByte(LastConfigByte);
+
+            
+            if (_lastChannel != channel)
+            {
+                Console.WriteLine($"Channel update failed from {_lastChannel} to {channel}");
```

Comment:
I'd recommend throwing or returning an exception instead. Writing to output is almost always not expected by user of the library

Reply:
I forgot this one too.
Dropped in dummy error list as input. Was useful to see when more than one errors were made on the conversion.